app/scripts/polling_location_finder.js

```diff
@@ -20,9 +29,165 @@ define(['jquery', 'geojson', 'json!vendor/ELECTIONS_WardsPrecincts.geojson', 'js
     function showResults() {
         $('.voting-result').show();
     }
+    
+    var weekDays = ["Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday"];
+    var monthNames = ["January", "February", "March", "April", "May", "June",
+                      "July", "August", "September", "October", "November", "December"];
+    
+    var pollListAccordionItems = new Array();
+    
+    function togglePollListItem() {
+        
+      var target = getAccordionItem(this.parentNode);
+      var itemClass = target.className
+      // Hide all items
+      for ( var i = 0; i < pollListAccordionItems.length; i++ ) {
+          var target_i = getAccordionItem(pollListAccordionItems[i]);
+          target_i.className = 'hide';
+      }
+
+      // Show this item if it was previously hidden
+      if ( itemClass == 'hide' ) {
+          target.className = '';
+      }        
+    }
+    
+    function getAccordionItem(item) {
+        for ( var i = 0; i < item.childNodes.length; i++ ) {
+            if (item.childNodes[i].id == 'open_hrs_div') {
+                return item.childNodes[i];
+            }
+        }
+    }    
 
     // keep track of user precinct across calls so we can erase previous precincts if necessary
     var userPrecinct;
+    
+    function formatEarlyVotingLocationListDisplayHours(hrs_ul, hours) {
+        
+        var innerHTML = '';
+        
+        moment.tz.add('America/New_York|EST EDT|50 40|0101|1Lz50 1zb0 Op0');
+        
+        var time_interval = ' to '        
+        
+        var today=new Date();
+        today.setHours(0);
+        today.setMinutes(0);
+        today.setSeconds(0);
+        
+        jQuery.each(hours, function(index, timeInterval) {
+            var range = moment.range(timeInterval);
+            var dates = range.toDate();
+            
+            var startDate = new Date(dates[0]);
+            var endDate = new Date(dates[1]);
+            var open_date = new Date(dates[0]);
+            open_date.setHours(0);
+            open_date.setMinutes(0);
+            open_date.setSeconds(0);            
+            
+            var li_class = ''
+            //Time zone is causing confusion, so resorted to brute force
+            if( open_date.getYear() < today.getYear() ) {
```

Comment:
@osesina moment allows you to feed it many different date formats. The ones in the geojson file work great with moment.js. We already have moment.range and moment as dependancies so it would be great to use them for this.